File: week2/KJW/BOJ_1159.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";

const [n, ...names] = fs.readFileSync(filePath).toString().trim().split("\n");
let obj = {};

for (let i = 0; i <= n - 1; i++) {
  if (!obj[names[i][0]]) {
    obj[names[i][0]] = 1;
  } else {
    obj[names[i][0]] += 1;
  }
}

const key_arr = Object.keys(obj);
const length = key_arr.length;
let result = "";
for (let j = 0; j <= length - 1; j++) {
  if (obj[key_arr[j]] >= 5) {
    result += key_arr[j];
  }
}

if (!result) {
  console.log("PREDAJA");
} else {
  let arr = result.split("");
  arr.sort();
  console.log(arr.join(""));
}

File: week02/hs/BOJ_1159.py
n = int(input())

change = {'a':0, 'b':0, 'c':0, 'd':0, 'e':0, 'f':0, 'g':0, 'h':0, 'i':0, 'j':0, 'k':0, 'l':0, 'm':0, 'n':0, 'o':0, 'p':0, 'q':0, 'r':0, 's':0, 't':0, 'u':0, 'v':0, 'w':0, 'x':0, 'y':0, 'z':0}

for i in range(n) :
    first = input()
    change[first[0]] += 1

result = []
for i in change:
    if change[i] >= 5:
        result.append(i)

if len(result) == 0 :
    print("PREDAJA")
else:
    print(''.join(result))